python/ray/experimental/state/state_cli.py

```diff
@@ -307,7 +308,7 @@ def format_list_api_output(
 ) -> str:
     if len(state_data) == 0:
         return "No resource in the cluster"
-    state_data = [dataclasses.asdict(state) for state in state_data]
+    state_data = [state.asdict() for state in state_data]
```

Comment:
we need to control the to dict routine here since pydantic and python dataclasses don't really work the same with `dataclasses.asdict`.

Reply:
I guess we should eventually use pydantic only. but sgtm. (cc @alanwguo )